src/raft.rs

```diff
@@ -617,17 +621,16 @@ impl<T: Storage> Raft<T> {
         } else {
             let term = self.raft_log.term(pr.next_idx - 1);
             let ents = self.raft_log.entries(pr.next_idx, self.max_msg_size);
-            if term.is_err() || ents.is_err() {
-                // send snapshot if we failed to get term or entries.
-                if !self.prepare_send_snapshot(&mut m, pr, to) {
+            if let (Ok(term), Ok(mut ents)) = (term, ents) {
```

Comment:
I prefer to use `match` rather than `if let ... else` since the former is fewer lines of code and the latter is just sugar for the former